File: app/src/main/java/com/fivesys/alphamanufacturas/fivesys/Views/Adapters/MenuAdapter.java
package com.fivesys.alphamanufacturas.fivesys.Views.Adapters;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.fivesys.alphamanufacturas.fivesys.R;


public class MenuAdapter extends RecyclerView.Adapter<MenuAdapter.ViewHolder> {
    private String[] titulos;
    private int[] imagenes;
    private OnItemClickListener listener;

    public MenuAdapter(String[] titulos, int[] imagenes, OnItemClickListener listener) {
        this.titulos = titulos;
        this.imagenes = imagenes;
        this.listener = listener;
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.cardview_menu, parent, false);
        return new ViewHolder(view);

    }

    @Override
    public void onBindViewHolder(ViewHolder holder, final int position) {
        holder.bind(titulos, imagenes, position, listener);
    }

    @Override
    public int getItemCount() {
        return titulos.length;
    }

    public class ViewHolder extends RecyclerView.ViewHolder {
        private ImageView imageViewPhoto;
        private TextView textViewTitulo;

        public ViewHolder(View itemView) {
            super(itemView);
            imageViewPhoto = itemView.findViewById(R.id.imageViewPhoto);
            textViewTitulo = itemView.findViewById(R.id.textViewTitulo);
        }

        private void bind(final String[] string, int[] imagenes, final int position, final OnItemClickListener listener) {
            imageViewPhoto.setImageResource(imagenes[position]);
            textViewTitulo.setText(string[position]);
            itemView.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View view) {
                    listener.onItemClick(string[position], getAdapterPosition());
                }
            });
        }
    }

    public interface OnItemClickListener {
        void onItemClick(String strings, int position);
    }
}

File: app/src/main/java/com/fivesys/alphamanufacturas/fivesys/views/adapters/MenuAdapter.kt
package com.fivesys.alphamanufacturas.fivesys.views.adapters

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.core.content.ContextCompat
import androidx.recyclerview.widget.RecyclerView
import com.fivesys.alphamanufacturas.fivesys.R
import com.google.android.material.card.MaterialCardView


class MenuAdapter(private val titulos: Array<String>, private val imagenes: IntArray, private val listener: OnItemClickListener) : RecyclerView.Adapter<MenuAdapter.ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view: View = LayoutInflater.from(parent.context).inflate(R.layout.cardview_menu, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.bind(titulos, imagenes, position, listener)
    }

    override fun getItemCount(): Int {
        return titulos.size
    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        private val card_view: MaterialCardView = itemView.findViewById(R.id.card_view)
        private val imageViewPhoto: ImageView = itemView.findViewById(R.id.imageViewPhoto)
        private val textViewTitulo: TextView = itemView.findViewById(R.id.textViewTitulo)

        fun bind(string: Array<String>, imagenes: IntArray, position: Int, listener: OnItemClickListener) {
            imageViewPhoto.setImageResource(imagenes[position])
            when (position) {
                0 -> {
                    card_view.setCardBackgroundColor(ContextCompat.getColor(itemView.context, R.color.colorAmarrillo))
                    textViewTitulo.text = string[position]
                    textViewTitulo.setTextColor(ContextCompat.getColor(itemView.context, R.color.colorPlomo))
                }
                1 -> {
                    card_view.setCardBackgroundColor(ContextCompat.getColor(itemView.context, R.color.colorWhite))
                    textViewTitulo.text = string[position]
                    textViewTitulo.setTextColor(ContextCompat.getColor(itemView.context, R.color.colorAzul))
                }
                2 -> {
                    card_view.setCardBackgroundColor(ContextCompat.getColor(itemView.context, R.color.colorAzulOscuro))
                    textViewTitulo.text = string[position]
                    textViewTitulo.setTextColor(ContextCompat.getColor(itemView.context, R.color.colorAmarrillo))
                }
            }

            itemView.setOnClickListener { listener.onItemClick(string[position], adapterPosition) }
        }
    }

    interface OnItemClickListener {
        fun onItemClick(strings: String, position: Int)
    }
}
